src/Features/Core/Portable/CodeFixesAndRefactorings/CodeActionRequestPriorityProvider.cs

```diff
@@ -81,25 +87,33 @@ public static bool MatchesPriority(this ICodeActionRequestPriorityProvider provi
             return true;
         }
 
-        if (provider.Priority == CodeActionRequestPriority.Low)
+        if (provider.Priority == codeFixProvider.RequestPriority)
+        {
+            return true;
+        }
+
+        if (provider.Priority == CodeActionRequestPriority.Low && provider.HasDeprioritizedAnalyzerSupportingDiagnosticId(codeFixProvider.FixableDiagnosticIds))
         {
             // 'Low' priority can be used for two types of code fixers:
             //  1. Those which explicitly set their 'RequestPriority' to 'Low' and
             //  2. Those which can fix diagnostics for expensive analyzers which were de-prioritized
             //     to 'Low' priority bucket to improve lightbulb population performance.
-            // Hence, when processing the 'Low' Priority bucket, we accept fixers with any RequestPriority,
-            // as long as they can fix a diagnostic from an analyzer that was executed in the 'Low' bucket.
+            // Hence, when processing the 'Low' Priority bucket and the priority provider indicates
```

Comment:
perhaps mention that the first case was handled by the prior if statement.  so we def have a case where the codefixprovider is higher priority than the fix priority itself.

(actually, it might be worthwhile to actually validate the codefixprovider priority here as well(

Reply:
Updated the comment and added a check to verify that the codeFixProvider's RequestPriority is either High or Default